tests/python/pants_test/reporting/test_reporting_remote_integration.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import re
+import subprocess
+import unittest
+from contextlib import contextmanager
+from typing import List
+
+from pants.util.collections import assert_single_element
+from pants.util.contextutil import http_server, temporary_dir
+from pants.util.reporting_util import (find_child_processes_that_send_spans, wait_spans_to_be_sent,
+                                       zipkin_handler)
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+class TestReportingRemoteIntegration(PantsRunIntegrationTest, unittest.TestCase):
+  @contextmanager
+  def run_mock_server(self, bin_name: str, arguments: List[str]):
+    """
+    Build and run a Rust mock server binary.
+    Mainly used for remoting integration tests.
+    It assumes that the server will print a line like `localhost:99999` to indicate its port.
+    :param bin_name: The name of the binary that you want to run. It must be in the main engine's workspace.
+    :param arguments: Passthrough arguments to pass to the binary.
+    """
+    args = ["./" + bin_name]
+    if len(arguments) > 0:
```

Comment:
If the list is non-optional, I'd probably just unconditionally `args.extend(arguments)`.

Reply:
If the list is empty (which it is for local_cas), it seems that python considers it None and extend doesn't work

Reply:
That's... Very surprising...

```
$ python3
Python 3.6.4 (default, Jan 23 2018, 10:58:43) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> l = ['a']
>>> l.extend([])
>>> l
['a']
```

I'm curious if you can repro this!

Reply:
Yes, this simple code snippet in the interpreter behaves the same for me. Not sure what's happening...